internal/gengapic/client_init.go

```diff
@@ -139,193 +139,405 @@ func (g *generator) clientOptions(serv *descriptor.ServiceDescriptorProto, servN
+	p("}")
+	p("")
 
-		if g.hasLocationMixin() {
-			p("  c.locationsClient = locationpb.NewLocationsClient(connPool)")
-			p("")
-		}
+	// setGoogleClientInfo method
+	p("// setGoogleClientInfo sets the name and version of the application in")
+	p("// the `x-goog-api-client` header passed on each request. Intended for")
+	p("// use by Google-written clients.")
+	p("func (c *%sGrpcClient) setGoogleClientInfo(keyval ...string) {", lowcaseServName)
+	p(`  kv := append([]string{"gl-go", versionGo()}, keyval...)`)
+	p(`  kv = append(kv, "gapic", versionClient, "gax", gax.Version, "grpc", grpc.Version)`)
+	p(`  c.xGoogMetadata = metadata.Pairs("x-goog-api-client", gax.XGoogHeader(kv...))`)
+	p("}")
+	p("")
 
-		p("  return c, nil")
-		p("}")
-		p("")
+	// Close method
+	p("// Close closes the connection to the API service. The user should invoke this when")
+	p("// the client is no longer required.")
+	p("func (c *%sGrpcClient) Close() error {", lowcaseServName)
+	p("  return c.connPool.Close()")
+	p("}")
+	p("")
+}
 
-		g.imports[pbinfo.ImportSpec{Name: "gtransport", Path: "google.golang.org/api/transport/grpc"}] = true
-		g.imports[pbinfo.ImportSpec{Path: "context"}] = true
+func (g *generator) makeClients(serv *descriptor.ServiceDescriptorProto, servName string) error {
+	var hasLRO bool
+	for _, m := range serv.Method {
+		if g.isLRO(m) {
+			hasLRO = true
+			break
+		}
 	}
 
-	// Connection()
-	{
-		p("// Connection returns a connection to the API service.")
-		p("//")
-		p("// Deprecated.")
-		p("func (c *%sClient) Connection() *grpc.ClientConn {", servName)
-		p("  return c.connPool.Conn()")
-		p("}")
-		p("")
+	imp, err := g.descInfo.ImportSpec(serv)
+	if err != nil {
+		return err
 	}
 
-	// Close()
-	{
-		p("// Close closes the connection to the API service. The user should invoke this when")
-		p("// the client is no longer required.")
-		p("func (c *%sClient) Close() error {", servName)
-		p("  return c.connPool.Close()")
-		p("}")
-		p("")
+	err = g.abstractClientIntfInit(serv, servName)
+	if err != nil {
+		return err
 	}
-
-	// setGoogleClientInfo
-	{
-		p("// setGoogleClientInfo sets the name and version of the application in")
-		p("// the `x-goog-api-client` header passed on each request. Intended for")
-		p("// use by Google-written clients.")
-		p("func (c *%sClient) setGoogleClientInfo(keyval ...string) {", servName)
-		p(`  kv := append([]string{"gl-go", versionGo()}, keyval...)`)
-		p(`  kv = append(kv, "gapic", versionClient, "gax", gax.Version, "grpc", grpc.Version)`)
-		p(`  c.xGoogMetadata = metadata.Pairs("x-goog-api-client", gax.XGoogHeader(kv...))`)
-		p("}")
-		p("")
+	g.clientInit(serv, servName, imp, hasLRO)
+
+	for _, v := range g.opts.transports {
+		switch v {
+		case grpc:
+			g.grpcClientInit(serv, servName, imp, hasLRO)
+		case rest:
+			// TODO(dovs): add rest client struct initialization
+			continue
+		default:
+			return fmt.Errorf("unexpected transport variant (supported variants are '%s', '%s'): %d",
```

Comment:
```suggestion
			return fmt.Errorf("unexpected transport variant (supported variants are %q, %q): %d",
```

The `%q` format specifier prints strings wrapped in double quotes.

Reply:
TIL